File: classes01/python/dilate.py
import cv2

imageName="../../data/images/truth.png"
image=cv2.imread(imageName,cv2.IMREAD_COLOR)

if image is None:
    print("图片加载失败")

dilationSize=6
element = cv2.getStructuringElement(cv2.MORPH_CROSS,(2*dilationSize+1,2*dilationSize+1),(dilationSize,dilationSize))

imageDialted=cv2.dilate(image,element)

cv2.imshow("Original",image)
cv2.imshow("Dilated Image",imageDialted)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: classes01/cpp/dilate.cpp
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>

using namespace std;
using namespace cv;

int main(void)
{
    Mat image;

    // 读取一张图片
    image=imread("../../data/images/truth.png", IMREAD_COLOR);

    // 创建一个核
    int dilationSize=6;
    Mat element=getStructuringElement(cv::MORPH_CROSS,cv::Size(2*dilationSize+1,2*dilationSize+1),cv::Point(dilationSize,dilationSize));

    // 定义个Mat用来装膨胀之后的图像
    Mat imageDilated;
    dilate(image,imageDilated,element);
    imshow("Original Image", image);
    imshow("Dilation",imageDilated);

    waitKey();
    
    return EXIT_SUCCESS;
}